syntax = "proto3";
package im;

option cc_enable_arenas = true;
option go_package = "./;im";

import "common.proto";

service AuthService {
    rpc Register (RegisterReq) returns (RegisterResp);
    rpc Login (LoginReq) returns (LoginResp);
    rpc Logout (LogoutReq) returns (LogoutResp);
}

message RegisterReq {
    string username = 1;
    string password = 2; 
}

message RegisterResp {
    bool success = 1;
    string error_msg = 2;
    uint64 user_id = 3;
}

message LoginReq {
    string username = 1;
    string password = 2;
}

message LoginResp {
    bool success = 1;
    string error_msg = 2;
    User user_info = 3;
    string token = 4;
}

message LogoutReq {
    string token = 1;
}

message LogoutResp {
    bool success = 1;
    string error_msg = 2;
}
